### PR TITLE
Do not error on maxUploadChange

### LINKED ISSUE #994: max file size-> save go to blanck page
### Steps to reproduce
1. go to the menu Administration/additional settings
2. go to file management and change the value of the max file (510 MB)
3. Click "Save"
4. a blanck page is displayed instead of the nextcloud page

### Server configuration
**Operating system**:
ubuntu 16.04.1

**Web server:**
nginx 1.10.1

**Database:**
mariaDB 10.0.25

**PHP version:**
PHP 7.0.8-0ubuntu0.16.04.2 (cli) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.8-0ubuntu0.16.04.2, Copyright (c) 1999-2016, by Zend Technologies

**Nextcloud version:** (see Nextcloud admin page)
Nextcloud 10.0 RC1 (testing)

**Updated from an older Nextcloud/ownCloud or fresh install:**
updated from 9.0.53

**Where did you install Nextcloud from:**
virtual server

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.

No errors have been found.

</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**The content of config/config.php:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here
(Without the database password, passwordsalt and secret)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no
yes
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';

Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**
Firefox 48

**Operating system:**
OSX

### Logs
#### Web server error log
<details>
<summary>Web server error log</summary>
```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>
```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

### PR DESCRIPTION
**Pull request by rullzer**
The new admin page broke the updating of the max upload size. This is
due to very old code there.
- Now update is done in the background using js
- Added proper route+controller to update

Fixes #994

CC: @Guillaume99 @blizzz @nickvergessen @MorrisJobke @icewind1991

### CODE REVIEW

**Comment by rullzer** [comment]
To test.

Before: :boom: 
Now: not(:boom:)

File: apps/files/js/admin.js [hunk 1]
```diff
@@ -8,6 +8,46 @@
...
+			});
+
+			request.fail(function () {
+				OC.msg.finishedError('#maxUploadSizeSettingsMsg', 'Error');
+			});
+		}
+	}
+})();
+
 function switchPublicFolder() {
```
Comment (blizzz) [inline apps/files/js/admin.js]: I believe the switchPublicFolder-related stuff can be totally removed, I did not find any other tracks or references in the code about it.

File: apps/files/templates/admin.php [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
-		<input type="hidden" value="<?php p($_['requesttoken']); ?>" name="requesttoken" />
-		<?php if($_['uploadChangable']): ?>
-			<input type="submit" name="submitFilesAdminSettings" id="submitFilesAdminSettings"
-				   value="<?php p($l->t( 'Save' )); ?>"/>
-			<p><em><?php p($l->t('With PHP-FPM it might take 5 minutes for changes to be applied.')); ?></em></p>
-		<?php else: ?>
-			<p><em><?php p($l->t('Missing permissions to edit from here.')); ?></em></p>
-		<?php endif; ?>
-	</form>
+	<h2><?php p($l->t('File handling')); ?></h2>
```
Comment (blizzz) [inline apps/files/templates/admin.php]: ``class="section"`` is missing, breaks layout

Comment (blizzz) [inline apps/files/templates/admin.php]: ah, needs to be in a div of course

**Comment by blizzz** [comment]
:+1: tested and works. Needs backport to stable10.

File: apps/files/js/admin.js (hunk 1 above)
Comment (rullzer) [inline apps/files/js/admin.js]: Yeah fair enough but lets do that seperatly ;)

File: apps/files/templates/admin.php (hunk 2 above)
Comment (rullzer) [inline apps/files/templates/admin.php]: Ah my bad. Thanks!

**Comment by blizzz** [comment]
Needs another reviewer @LukasReschke @mar1u5

**Comment by LukasReschke** [comment]
LGTM

**Comment by MorrisJobke** [comment]
@blizzz Same here. Please remove the backport-request label once the backport PR is opened. Makes it easier to spot the not yet backported PRs ;)

**Pull request merged**